From 4ab46f26c1f020f5867ee028befe517b5c544b15 Mon Sep 17 00:00:00 2001
From: Alexander Schlarb <alexander@ninetailed.ninja>
Date: Fri, 13 Dec 2019 11:48:58 +0100
Subject: [PATCH] contrib: Use system GMP if its header is present

---
 contrib/src/gmp/rules.mak | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/contrib/src/gmp/rules.mak b/contrib/src/gmp/rules.mak
index 77f2cfc1d..9218710af 100644
--- a/daemon/contrib/src/gmp/rules.mak
+++ b/daemon/contrib/src/gmp/rules.mak
@@ -3,6 +3,12 @@
 GMP_VERSION := 6.1.2
 GMP_URL := $(GNU)/gmp/gmp-$(GMP_VERSION).tar.bz2
 
+ifndef HAVE_CROSS_COMPILE
+ifeq ($(shell echo '\#include <gmp.h>' | $(CC) -E - >/dev/null 2>&1 || echo FAIL),)
+PKGS_FOUND += gmp
+endif
+endif
+
 $(TARBALLS)/gmp-$(GMP_VERSION).tar.bz2:
 	$(call download,$(GMP_URL))
 
-- 
2.24.0

