--- ring-client-gnome/scripts/install.sh
+++ ring-client-gnome/scripts/install.sh
@@ -37,14 +37,14 @@ while getopts gsc:p: OPT; do
 done

 make_install() {
-  if $1; then
-    sudo make install
-    # Or else the next non-sudo install will fail, because this generates some
-    # root owned files like install_manifest.txt under the build directory.
-    sudo chown -R "$USER" .
-  else
+  #if $1; then
+  #  sudo make install
+  #  # Or else the next non-sudo install will fail, because this generates some
+  #  # root owned files like install_manifest.txt under the build directory.
+  #  sudo chown -R "$USER" .
+  #else
     make install
-  fi
+  #fi
 }

 TOP="$(pwd)"
@@ -61,7 +61,7 @@
 cd contrib
 mkdir -p native
 cd native
-../bootstrap
+../bootstrap --disable-downloads --disable-ffmpeg --disable-gcrypt --disable-gmp --disable-gnutls --disable-gpg-error --disable-nettle --disable-opus --disable-speex --disable-speexdsp --disable-vpx --disable-zlib --disable-libressl --enable-debug --prefix=/app
 make
 cd "${DAEMON}"
 ./autogen.sh
@@ -75,18 +75,18 @@ if [[ "$OSTYPE" != "darwin"* ]]; then
 fi

 if $global; then
-  ./configure $sharedLib $CONFIGURE_FLAGS
+  ./configure $sharedLib $CONFIGURE_FLAGS --prefix="/app"
 else
   ./configure $sharedLib $CONFIGURE_FLAGS --prefix="${INSTALL}/daemon"
 fi
 make -j${proc}
-make_install $global
+make_install

 cd "${TOP}/lrc"
 mkdir -p ${BUILDDIR}
 cd ${BUILDDIR}
 if $global; then
-  cmake .. -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH -DCMAKE_BUILD_TYPE=Debug $static
+  cmake .. -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX="/app" $static
 else
   cmake ..  -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH \
             -DCMAKE_BUILD_TYPE=Debug \
@@ -100,7 +100,7 @@ else
 mkdir -p ${BUILDDIR}
 cd ${BUILDDIR}
 if $global; then
-  cmake .. -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH $static
+  cmake .. -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH -DCMAKE_INSTALL_PREFIX="/app" $static
 else
   cmake ..  -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH \
             -DCMAKE_INSTALL_PREFIX="${INSTALL}/${client}" \