diff --git a/src/main.cpp b/src/main.cpp
index e8ab280..659b75f 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -62,7 +62,6 @@ main(int argc, char* argv[])
     QList<char*> qtWebEngineChromiumFlags;
 
 #ifdef Q_OS_LINUX
-    setenv("QT_QPA_PLATFORMTHEME", "gtk3", true);
     setenv("QML_DISABLE_DISK_CACHE", "1", true);
 
     /*
diff --git a/src/mainapplication.cpp b/src/mainapplication.cpp
index 918200b..bafa23b 100644
--- a/src/mainapplication.cpp
+++ b/src/mainapplication.cpp
@@ -170,11 +170,6 @@ MainApplication::init()
     QResource::registerResource(QCoreApplication::applicationDirPath() + QDir::separator()
                                 + "webresource.rcc");
 
-#ifdef Q_OS_LINUX
-    if (!getenv("QT_QPA_PLATFORMTHEME"))
-        setenv("QT_QPA_PLATFORMTHEME", "gtk3", true);
-#endif
-
     auto results = parseArguments();
 
     if (results[opts::DEBUG].toBool()) {
