diff --git a/src/main.cpp b/src/main.cpp
index a0b923e..6cd9284 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -63,8 +63,10 @@ main(int argc, char* argv[])
 
 #ifdef Q_OS_LINUX
     if (!getenv("QT_QPA_PLATFORMTHEME")
-        && !(getenv("XDG_CURRENT_DESKTOP") == "KDE" || getenv("XDG_CURRENT_DESKTOP") == "GNOME"))
+        && !(strcmp(getenv("XDG_CURRENT_DESKTOP"), "KDE") == 0
+        || strcmp(getenv("XDG_CURRENT_DESKTOP"), "GNOME") == 0)) {
         setenv("QT_QPA_PLATFORMTHEME", "gtk3", true);
+    }
     setenv("QML_DISABLE_DISK_CACHE", "1", true);
 
     /*
