diff --git a/src/main.cpp b/src/main.cpp
index d401027..f606a6b 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -62,7 +62,6 @@ main(int argc, char* argv[])
     QList<char*> qtWebEngineChromiumFlags;
 
 #ifdef Q_OS_LINUX
-    setenv("QT_QPA_PLATFORMTHEME", "gtk3", true);
     setenv("QML_DISABLE_DISK_CACHE", "1", true);
 
     /*
diff --git a/src/mainapplication.cpp b/src/mainapplication.cpp
index 02616e9..69158f9 100644
--- a/src/mainapplication.cpp
+++ b/src/mainapplication.cpp
@@ -166,7 +166,8 @@ MainApplication::init()
     setWindowIcon(QIcon(":/images/jami.ico"));
 
 #ifdef Q_OS_LINUX
-    if (!getenv("QT_QPA_PLATFORMTHEME"))
+    if (!(getenv("XDG_CURRENT_DESKTOP") == "KDE" ||
+        getenv("XDG_CURRENT_DESKTOP") == "GNOME"))
         setenv("QT_QPA_PLATFORMTHEME", "gtk3", true);
 #endif
 
