From 6bb307bda4c74b78184d454d0dca9bfd9e11df34 Mon Sep 17 00:00:00 2001
From: Vladimir Stoiakin <VStoiakin@lavabit.com>
Date: Thu, 04 Nov 2021 13:49:57 +0300
Subject: [PATCH] meson: update the list of sources and dependencies

Change-Id: I6b419ff3e1bc4f8dd8958974e4ae4bc98d30cf06
---

diff --git a/meson.build b/meson.build
index c9f688b..7f349e2 100644
--- a/meson.build
+++ b/meson.build
@@ -14,6 +14,7 @@
 depthreads = dependency('threads')
 depopendht = dependency('opendht', version: '>= 2.1.0')
 depgnutls = dependency('gnutls', version: '>= 3.6.7')
+depnettle = dependency('nettle', version: '>= 3.0.0')
 deplibpjproject = dependency('libpjproject')
 deplibgit2 = dependency('libgit2', version: '>= 1.1.0')
 deplibsecp256k1 = dependency('libsecp256k1', version: '>= 0.1')
@@ -129,17 +130,21 @@
 
 if get_option('plugins')
     conf.set('ENABLE_PLUGIN', true)
-    deplibarchive = dependency('libarchive', version: '>= 3.4.0', required: false)
-    if not deplibarchive.found()
-        deplibarchive = modcmake.subproject('libarchive', cmake_options: [
-            '-DCMAKE_BUILD_TYPE=Release',
-            '-DCMAKE_POSITION_INDEPENDENT_CODE=ON',
-            '-DENABLE_TEST=OFF',
-            '-DENABLE_TAR=OFF',
-            '-DENABLE_CPIO=OFF',
-            '-DENABLE_CAT=OFF',
-            '-DENABLE_LIBXML2=OFF'
-        ]).dependency('archive_static')
+    if host_machine.system() == 'darwin'
+        depminizip = dependency('minizip', version: '>= 3.0.0')
+    else
+        deplibarchive = dependency('libarchive', version: '>= 3.4.0', required: false)
+        if not deplibarchive.found()
+            deplibarchive = modcmake.subproject('libarchive', cmake_options: [
+                '-DCMAKE_BUILD_TYPE=Release',
+                '-DCMAKE_POSITION_INDEPENDENT_CODE=ON',
+                '-DENABLE_TEST=OFF',
+                '-DENABLE_TAR=OFF',
+                '-DENABLE_CPIO=OFF',
+                '-DENABLE_CAT=OFF',
+                '-DENABLE_LIBXML2=OFF'
+            ]).dependency('archive_static')
+        endif
     endif
     depdl = meson.get_compiler('cpp').find_library('dl', required: false)
 else
diff --git a/src/meson.build b/src/meson.build
index 1fd89a7..991ea85 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -39,6 +39,7 @@
     'jamidht/server_account_manager.cpp',
     'jamidht/sync_channel_handler.cpp',
     'jamidht/sync_module.cpp',
+    'jamidht/transfer_channel_handler.cpp',
     'media/audio/echo-cancel/null_echo_canceller.cpp',
     'media/audio/sound/audiofile.cpp',
     'media/audio/sound/dtmf.cpp',
@@ -130,6 +131,7 @@
     depthreads,
     depopendht,
     depgnutls,
+    depnettle,
     deplibpjproject,
     deplibgit2,
     deplibsecp256k1,
@@ -298,7 +300,12 @@
         'plugin/pluginpreferencesutils.cpp',
         'plugin/pluginsutils.cpp'
     )
-    libjami_dependencies += [deplibarchive, depdl]
+    if host_machine.system() == 'darwin'
+        libjami_dependencies += depminizip
+    else
+        libjami_dependencies += deplibarchive
+    endif
+    libjami_dependencies += depdl
 endif
 
 libjami = library('jami',
