diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29b9f1c2..a981ba01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -384,6 +384,7 @@ SET(libringclient_WEB_chatview
    src/web-chatview/chatview.js
    src/web-chatview/jed.js
    src/web-chatview/emoji.js
+   src/web-chatview/emoji.css
    src/web-chatview/linkify-html.js
    src/web-chatview/linkify-string.js
    src/web-chatview/linkify.js
