diff --git a/CMakeLists.txt b/CMakeLists.txt
index b73781a0c..48c9ae026 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -168,6 +168,7 @@ if(NOT MSVC)
             message(FATAL_ERROR "No audio library found. Please install PulseAudio, ALSA or JACK.")
         endif()
     endif()
+    list(APPEND PKG_CONFIG_EXECUTABLE "--static")
 endif()
 
 ################################################################################
@@ -296,7 +297,7 @@ elseif(MSVC)
    endif()
 endif()
 
-add_library(${PROJECT_NAME} STATIC ${ALL_FILES})
+add_library(${PROJECT_NAME} SHARED ${ALL_FILES})
 
 set_target_properties(${PROJECT_NAME} PROPERTIES FOLDER "daemon")
 target_compile_definitions(${PROJECT_NAME} PRIVATE
@@ -311,6 +312,7 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE
 target_compile_definitions(${PROJECT_NAME} PUBLIC
     ASIO_STANDALONE
     MSGPACK_NO_BOOST
+    jami_EXPORTS
 )
 if (JAMI_VIDEO)
     target_compile_definitions(${PROJECT_NAME} PRIVATE ENABLE_VIDEO)
@@ -607,7 +609,7 @@ else()
         target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::udev)
         if (pulseaudio_FOUND)
             target_compile_definitions(${PROJECT_NAME} PRIVATE HAVE_PULSE)
-            target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::pulseaudio)
+            target_link_libraries(${PROJECT_NAME} PUBLIC PkgConfig::pulseaudio)
         endif()
         if (alsa_FOUND)
             target_compile_definitions(${PROJECT_NAME} PRIVATE HAVE_ALSA)