diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf6012ec..e9364269 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,13 +71,6 @@ set(CLIENT_LIBS, "")
 include(${PROJECT_SOURCE_DIR}/extras/build/cmake/contrib_tools.cmake)
 set(EXTRA_PATCHES_DIR ${PROJECT_SOURCE_DIR}/extras/patches)
 
-list(APPEND QWINDOWKIT_OPTIONS
-  QWINDOWKIT_BUILD_WIDGETS OFF
-  QWINDOWKIT_INSTALL OFF
-  QWINDOWKIT_BUILD_STATIC ON
-  QWINDOWKIT_BUILD_QUICK ON
-)
-
 if(WIN32)
   # Beta config
   if(BETA)
@@ -87,26 +80,8 @@ if(WIN32)
   set(JAMI_OUTPUT_DIRECTORY_RELEASE ${PROJECT_SOURCE_DIR}/x64/${CMAKE_BUILD_TYPE})
 endif()
 
-if(WIN32)
-  list(APPEND QWINDOWKIT_OPTIONS QWINDOWKIT_ENABLE_WINDOWS_SYSTEM_BORDERS OFF)
-endif()
+find_package(QWindowKit REQUIRED)
 
-# If qwindowkit is unable to find qmsetup via cmake's find_package function, it will install it and
-# then call find_package again. Unfortunately, even the second call to find_package sometimes
-# fails due to qmsetup having been installed in the wrong directory. The following patch
-# ensures that qmsetup is always installed in the directory where find_package looks for it.
-if(NOT WIN32)
-  list(APPEND QWINDOWKIT_PATCHES ${EXTRA_PATCHES_DIR}/0001-fix-qm_install_package-function.patch)
-endif()
-
-# qwindowkit (frameless window)
-add_fetch_content(
-  TARGET qwindowkit
-  URL https://github.com/stdware/qwindowkit.git
-  BRANCH 758b00cb6c2d924be3a1ea137ec366dc33a5132d
-  PATCHES ${QWINDOWKIT_PATCHES}
-  OPTIONS ${QWINDOWKIT_OPTIONS}
-)
 list(APPEND CLIENT_INCLUDE_DIRS ${QWindowKit_BINARY_DIR}/include)
 list(APPEND CLIENT_LIBS QWindowKit::Quick)