diff --git a/src/app/settingsview/components/LinkedDevicesBase.qml b/src/app/settingsview/components/LinkedDevicesBase.qml
index 55bdad1..b8c1606 100644
--- a/src/app/settingsview/components/LinkedDevicesBase.qml
+++ b/src/app/settingsview/components/LinkedDevicesBase.qml
@@ -18,10 +18,10 @@
 import QtQuick.Controls
 import QtQuick.Layouts
 import SortFilterProxyModel 0.2
+import net.jami.Enums 1.1
 import net.jami.Models 1.1
 import net.jami.Adapters 1.1
 import net.jami.Constants 1.1
-import net.jami.Enums 1.1
 import "../../commoncomponents"
 
 ColumnLayout {
diff --git a/src/app/wizardview/components/NoUsernamePopup.qml b/src/app/wizardview/components/NoUsernamePopup.qml
index be5b376..f371772 100644
--- a/src/app/wizardview/components/NoUsernamePopup.qml
+++ b/src/app/wizardview/components/NoUsernamePopup.qml
@@ -55,7 +55,7 @@
         Accessible.role: Accessible.Dialog
         Accessible.name: title
         Accessible.description: JamiStrings.joinJamiNoPassword
-        width: root.width - 2 * root.popupMargins
+        width: 400 - 2 * popupMargins
         font.pixelSize: JamiTheme.popuptextSize
 
         lineHeight: JamiTheme.wizardViewTextLineHeight