From 6cc447a7c5a943dda27e99f0e41ede33957bae54 Mon Sep 17 00:00:00 2001
From: Aline Gondim Santos <aline.gondimsantos@savoirfairelinux.com>
Date: Tue, 26 Sep 2023 12:48:10 -0300
Subject: [PATCH] misc: fix start without existing profile

Change-Id: If6fe686345fc9e54c729c99766f9eec6ea382306
GitLab: #1366
---

diff --git a/src/app/currentaccount.cpp b/src/app/currentaccount.cpp
index 0867aa6..78e9cf0 100644
--- a/src/app/currentaccount.cpp
+++ b/src/app/currentaccount.cpp
@@ -90,6 +90,10 @@
 void
 CurrentAccount::setupForAccount()
 {
+    if (!lrcInstance_->getCurrentContactModel()) {
+        qWarning() << Q_FUNC_INFO << "No contact model";
+        return;
+    }
     connect(lrcInstance_->getCurrentContactModel(),
             &ContactModel::bannedStatusChanged,
             this,
