--- indicator/configure.ac.orig	2018-03-21 18:39:42.000000000 +0100
+++ indicator/configure.ac	2019-04-10 21:11:53.288946606 +0200
@@ -45,7 +45,6 @@
 GTK_REQUIRED_VERSION=2.18
 GTK3_REQUIRED_VERSION=3.6
 GIO_UNIX_REQUIRED_VERSION=2.37
-IDO_REQUIRED_VERSION=13.10.0
 
 AC_ARG_WITH([gtk],
   [AS_HELP_STRING([--with-gtk],
@@ -55,8 +54,7 @@
 AS_IF([test "x$with_gtk" = x3],
         [PKG_CHECK_MODULES(LIBINDICATOR,  gtk+-3.0 >= $GTK3_REQUIRED_VERSION
                                           gmodule-2.0
-                                          gio-unix-2.0 >= $GIO_UNIX_REQUIRED_VERSION
-                                          libido3-0.1 >= $IDO_REQUIRED_VERSION)
+                                          gio-unix-2.0 >= $GIO_UNIX_REQUIRED_VERSION)
         ],
       [test "x$with_gtk" = x2],
         [PKG_CHECK_MODULES(LIBINDICATOR,  gtk+-2.0 >= $GTK_REQUIRED_VERSION
@@ -160,7 +158,6 @@
 libindicator/indicator-0.4.pc.in
 libindicator/indicator3-0.4.pc.in
 tests/Makefile
-tools/Makefile
 ])
 AC_OUTPUT
 
--- libindicator3/Makefile.am.orig	2018-03-21 18:39:42.000000000 +0100
+++ libindicator3/Makefile.am	2019-04-10 21:15:18.617333776 +0200
@@ -2,8 +2,7 @@
 
 SUBDIRS = \
 	data \
-	libindicator \
-	tools
+	libindicator
 
 tools: libindicator
 
